client/lib/text-utils/index.js

```diff
@@ -24,4 +30,26 @@ function countWords( content ) {
 	return 0;
 }
 
-export default { countWords };
+const diffWords = JsDiff.diffWords;
+
+function countDiffWords( diffChanges ) {
+	return reduce(
+		diffChanges,
+		( accumulator, change ) => {
+			const count = countWords( change.value );
+			change.added && ( accumulator.added += count );
```

Comment:
Minor: I found myself staring at this function for minutes convinced we'd be inaccurately incrementing _both_ `added` and `removed` for each change, until I noticed that the first half of each line is treated effectively as a condition. This is one case where I'd advocate for the less magical `if ( change.added ) {` variation.